test/unit/app/app.test.js

```diff
@@ -626,6 +630,97 @@ describe('App',  () => {
+      }); 
+
+      const clinicianupload = {
+        allUsersMap: {
+          a1b2c3: {
+            termsAccepted: 'today'
+          },
+          d4e5f6: {}
+        },
+        currentPatientInViewId: 'd4e5f6',
+        loggedIn: true,
+        loggedInUserId: 'a1b2c3',
+        notification: {
+          key: 'fetchingPatient',
+          link: {
+            to: '/patients/foo',
+            text: 'Sorry!'
+          },
+          status: 405
+        },
+        permissionsOfMembersInTargetCareTeam: {},
+        membershipPermissionsInOtherCareTeams: {
+          d4e5f6: {
+            view: {},
+            note: {},
+            upload: {},
+          }
+        },
+        dataDonationAccounts: [],
+        datasources: [],
+        showingDonateBanner: null,
+        showingDexcomConnectBanner: null,
+        working: {
+          fetchingUser: {inProgress: false},
+          fetchingPendingSentInvites: {inProgress: false},
+          updatingDataDonationAccounts: {inProgress: false},
+          fetchingPatient: {inProgress: false, notification: {type: 'error'}},
+          loggingOut: {inProgress: false}
+        }
+      };
+      const clinicianuploadresult = mapStateToProps({blip: clinicianupload});
+
+      it('should return user is clinician with upload permissions', () => {
+        expect(clinicianuploadresult.permsOfLoggedInUser).to.equal(clinicianupload.membershipPermissionsInOtherCareTeams.d4e5f6);
+      });
+
+      const cliniciannoupload = {
+        allUsersMap: {
+          a1b2c3: {
+            termsAccepted: 'today'
+          },
+          d4e5f6: {}
+        },
+        currentPatientInViewId: 'd4e5f6',
+        loggedIn: true,
+        loggedInUserId: 'a1b2c3',
+        notification: {
+          key: 'fetchingPatient',
+          link: {
+            to: '/patients/foo',
+            text: 'Sorry!'
+          },
+          status: 405
+        },
+        permissionsOfMembersInTargetCareTeam: {},
+        membershipPermissionsInOtherCareTeams: {
+          d4e5f6: {
+            view: {},
+            note: {},
+          }
+        },
+        dataDonationAccounts: [],
+        datasources: [],
+        showingDonateBanner: null,
+        showingDexcomConnectBanner: null,
+        working: {
+          fetchingUser: {inProgress: false},
+          fetchingPendingSentInvites: {inProgress: false},
+          updatingDataDonationAccounts: {inProgress: false},
+          fetchingPatient: {inProgress: false, notification: {type: 'error'}},
+          loggingOut: {inProgress: false}
+        }
+      };
+      const cliniciannouploadresult = mapStateToProps({blip: cliniciannoupload});
+
+      it('should return user is clinician without upload permissionss', () => {
+        expect(cliniciannouploadresult.permsOfLoggedInUser).to.equal(cliniciannoupload.membershipPermissionsInOtherCareTeams.d4e5f6);
+      });
+
+      it('should return user is not clinician with empty permsOfLoggedInUser', () => {
```

Comment:
Just a note: After refactoring with context blocks, this should remain in the parent block, as it's related to that setup, and should remain in that scope.

Also, the it statement is misleading here, as the fact that the user does non have any permissions to see the current patient in view doesn't mean it's a clinician.

Perhaps 
```javascript
it('should return `null` for `permsOfLoggedInUser` when the user does not have any authorized access', () => {})